app/src/main/java/com/github/se/eventradar/viewmodel/CreateEventViewModel.kt

```diff
@@ -0,0 +1,352 @@
+package com.github.se.eventradar.viewmodel
+
+import android.net.Uri
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.github.se.eventradar.model.Location
+import com.github.se.eventradar.model.Resource
+import com.github.se.eventradar.model.User
+import com.github.se.eventradar.model.event.Event
+import com.github.se.eventradar.model.repository.event.IEventRepository
+import com.github.se.eventradar.model.repository.location.ILocationRepository
+import com.github.se.eventradar.model.repository.user.IUserRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import java.text.SimpleDateFormat
+import java.util.Locale
+import javax.inject.Inject
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+
+@HiltViewModel
+class CreateEventViewModel
+@Inject
+constructor(
+    private val locationRepository: ILocationRepository,
+    private val eventRepository: IEventRepository,
+    private val userRepository: IUserRepository
+) : ViewModel() {
+  private val _uiState = MutableStateFlow(CreateEventUiState())
+  val uiState: StateFlow<CreateEventUiState> = _uiState
+
+  /*
+  This function adds a new event based on the user input.
+  It firstly generates a new eventId.
+  Next, it uploads the event photo given by the user onto firestore and retrieves the link to this image.
+  Afterward, a hashmap is generated based on the user information.
+  Finally, the event firestore database is updated with the new event, and the user's event_hosting list is updated.
+   */
+  fun addEvent(state: MutableStateFlow<CreateEventUiState> = _uiState) {
+    viewModelScope.launch {
+      userRepository.getCurrentUserId().let { userIdResource ->
+        when (userIdResource) {
+          is Resource.Success -> {
+            val newEventId =
+                when (val result = eventRepository.getUniqueEventId()) {
+                  is Resource.Success -> {
+                    result.data
+                  }
+                  is Resource.Failure -> {
+                    Log.d("CreateEventViewModel", "Error Generating Event Id")
+                    return@launch
```

Comment:
without an error state, there's no way to know if addEvent was successful outside of this function.